SYCL/ESIMD/lsc/lsc_load2d_smoke.cpp

```diff
@@ -0,0 +1,90 @@
+//==------------ load2d_smoke.cpp - DPC++ ESIMD on-device test -------------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+// REQUIRES: gpu-intel-pvc || esimd_emulator
+// UNSUPPORTED: cuda || hip
+// RUN: %clangxx -fsycl %s -o %t.out
+// RUN: %GPU_RUN_PLACEHOLDER %t.out
+
+// Basic test for new lsc_load2d API.
+
+#include <algorithm>
+#include <cmath>
+#include <numeric>
+#include <random>
+#include <sycl/ext/intel/esimd.hpp>
+#include <sycl/sycl.hpp>
+
+int main() {
+  using namespace sycl;
+  using namespace sycl::ext::intel::esimd;
+  using namespace sycl::ext::intel::experimental::esimd;
+  constexpr uint32_t SurfaceHeight = 4;
+  constexpr uint32_t SurfaceWidth = 9;
+  constexpr uint32_t SurfacePitch = 16;
+  constexpr uint32_t x = 0;
+  constexpr uint32_t y = 0;
+  constexpr uint32_t BlockWidth = 4;
+  constexpr uint32_t BlockHeight = 4;
+  constexpr uint32_t NumBlocks = 1;
+
+  constexpr uint32_t Size = SurfaceHeight * SurfacePitch;
+
+  queue q;
+  auto device = q.get_device();
+  std::cout << "Device name: " << device.get_info<sycl::info::device::name>()
+            << std::endl;
+
+  auto *input = malloc_shared<int>(Size, q);
+  std::iota(input, input + Size, 0);
+
+  auto *block_store = malloc_shared<int>(Size, q);
+
+  auto *ref = new int[Size];
+
+  for (int i = 0; i < Size; i++)
+    block_store[i] = ref[i] = rand() % 128;
+
+  for (int i = 0; i < BlockHeight; i++) {
+    for (int j = 0; j < BlockWidth; j++) {
+      ref[y * SurfacePitch + i * SurfacePitch + x + j] =
+          input[y * SurfacePitch + i * SurfacePitch + x + j];
+    }
+  }
+  try {
+    q.submit([&](handler &h) {
+      h.parallel_for<class SimplestKernel>(
+          range<1>{1}, [=](id<1> id) SYCL_ESIMD_KERNEL {
+            config2d_mem_access<int, BlockWidth, BlockHeight, NumBlocks>
+                payload(input, (SurfaceWidth * sizeof(int)) - 1,
+                        SurfaceHeight - 1, (SurfacePitch * sizeof(int)) - 1, x,
+                        y);
+
+            auto data = lsc_load2d(payload);
+
+            payload.set_data_pointer(block_store);
+
+            lsc_store2d(payload, data);
+          });
+    });
+    q.wait();
+  } catch (sycl::exception e) {
+    std::cout << "SYCL exception caught: " << e.what();
+    free(input, q);
+    free(block_store, q);
+    return 1;
+  }
+
+  auto error = 0;
+  for (auto i = 0; i < Size; ++i)
+    error += std::abs(ref[i] - block_store[i]);
```

Comment:
it is matter of probability here, but there is non-zero chance that sum of differences may overflow 'int error' such a way that it would become equal to 0.
This more simpler code would work always:
```suggestion
    if (ref[i] != block_store[i])
      error++;
```

Reply:
Fixed